File: LibreriaV3.1/LibreriaV3.1/Negocio/TFactura.cs
using LibreriaV3._1.Comun;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibreriaV3._1.Negocio
{
    class TFactura
    {
        public string CodFactura { get; set; }

        public string Cliente { get; set; }

        public string FechaFactura { get; set; }

        public string Borrado { get; set; }

        public TFactura(string codFactura, string cliente, string fechaFactura, string borrado)
        {
            CodFactura = codFactura;
            Cliente = cliente;
            FechaFactura = fechaFactura;
            Borrado = borrado;
        }

        public TFactura() { }

        public TFactura(string cliente, string fechaFactura)
        {
            CodFactura = Util.GenerarCodigo(this.GetType());
            Cliente = cliente;
            FechaFactura = fechaFactura;
            Borrado = "0";
        }

        public override string ToString()
        {
            return CodFactura + " - " + Cliente;
        }

    }
}

File: LibreriaV6.1/src/Negocio/TFactura.java
package Negocio;

import Persistencia.UtilFichero;
import Persistencia.UtilSQL;

import javax.naming.NamingException;
import java.io.IOException;
import java.sql.SQLException;
import java.text.SimpleDateFormat;
import java.util.Calendar;

/**
 * Created by JAVI on 19/12/2016.
 */
public class TFactura {

    private String codFactura;
    private String usuario;
    private String fechaFactura;
    private String borrado;

    public TFactura() {
    }

    public TFactura(String codFactura, String usuario, String fechaFactura, String borrado) {
        this.codFactura = codFactura;
        this.usuario = usuario;
        this.fechaFactura = fechaFactura;
        this.borrado = borrado;
    }

    public TFactura(String usuario) throws ClassNotFoundException, SQLException, IOException, NamingException {
        Calendar calendario=Calendar.getInstance();
        SimpleDateFormat formato = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        this.codFactura= UtilSQL.generarCodigo(this.getClass());
        this.usuario = usuario;
        this.fechaFactura=formato.format(calendario.getTimeInMillis());
        this.borrado="0";
    }


    public String getCodFactura() {
        return codFactura;
    }

    public void setCodFactura(String codFactura) {
        this.codFactura = codFactura;
    }

    public String getUsuario() {
        return usuario;
    }

    public void setUsuario(String usuario) {
        this.usuario = usuario;
    }

    public String getFechaFactura() {
        return fechaFactura;
    }

    public void setFechaFactura(String fechaFactura) {
        this.fechaFactura = fechaFactura;
    }

    public String getBorrado() {
        return borrado;
    }

    public void setBorrado(String borrado) {
        this.borrado = borrado;
    }

    @Override
    public String toString() {
        return codFactura + " " + usuario + " " + fechaFactura + " " + borrado;
    }
}


